lisp/ess-inf.el

```diff
@@ -1992,6 +1990,8 @@ for `ess-eval-region'."
     (define-key map "\C-c\C-d"   'ess-doc-map)
     (define-key map "\C-c\C-e"   'ess-extra-map)
     (define-key map "\C-c\C-t"   'ess-dev-map)
+    (when ess-smart-S-assign-key
```

Comment:
This gives me a strange feeling because it seems people should use standard ways of settings keybindings instead. But in this case I guess it's a good compromise...

I wonder if we should have some sort of parent map that both `ess-mode-map` and `inferior-ess-mode-map` would inherit from. The common keybindings would be mapped there and this way you only need to customise one place.

Reply:
I agree it's a bit odd. Other modes do define them separately, though. For example, there's `python-mode`, derived from `prog-mode`, and `inferior-python-mode`, derived from `comint-mode`.

Reply:
oh yeah, that is the reason, thanks!